tests/test_datetime.py

```diff
@@ -112,53 +109,105 @@ def test_datetime_properties() -> None:
             datetime(2020, 4, 20, 15, 10, 33, 976508, 569718000, 529850102),
             "2020, 4, 20, 15, 10, 33, 976508, 569718000, 529850102",
         ),
+        (
+            datetime(2020, 4, 21, tzinfo=tzinfo(hours=1)),
+            "2020, 4, 21, 0, 0, "
+            "tzinfo=datetime.timezone(datetime.timedelta(seconds=3600))",
+        ),
+        (
+            datetime(1, 1, 1, ys=4, tzinfo=tzinfo(hours=-1)),
+            "1, 1, 1, 0, 0, 0, 0, 0, 4, "
+            "tzinfo=datetime.timezone(datetime.timedelta(days=-1, seconds=82800))",
+        ),
+        (
+            datetime(1, 1, 1, ys=4, tzinfo=tzinfo(hours=-1), fold=1),
+            "1, 1, 1, 0, 0, 0, 0, 0, 4, "
+            "tzinfo=datetime.timezone(datetime.timedelta(days=-1, seconds=82800)), "
+            "fold=1",
+        ),
+        (datetime(2020, 4, 21, fold=1), "2020, 4, 21, 0, 0, fold=1"),
+        (datetime(1, 1, 1, ys=4, fold=1), "1, 1, 1, 0, 0, 0, 0, 0, 4, fold=1"),
     ],
 )
 def test_datetime_repr(dt: hightime.datetime, middle_part: str) -> None:
+    import datetime as datetime  # noqa: F401 - for tzinfo eval
+
     assert repr(dt) == "hightime.datetime({})".format(middle_part)
     assert dt == eval(repr(dt))
-    # @TODO: tzinfo and fold
 
 
 @pytest.mark.parametrize(
-    "dt, expected",
+    "dt, expected, expected_tz",
     [
-        (datetime(2020, 4, 21, 15, 29, 34), "2020-04-21T15:29:34"),
-        (datetime(1, 1, 1, us=12345), "0001-01-01T00:00:00.012345"),
-        (datetime(1, 1, 1, us=10), "0001-01-01T00:00:00.000010"),
-        (datetime(1, 1, 1, fs=40), "0001-01-01T00:00:00.000000000000040"),
+        (datetime(2020, 4, 21, 15, 29, 34), "2020-04-21T15:29:34", ""),
+        (datetime(1, 1, 1, us=12345), "0001-01-01T00:00:00.012345", ""),
+        (datetime(1, 1, 1, us=10), "0001-01-01T00:00:00.000010", ""),
+        (datetime(1, 1, 1, fs=40), "0001-01-01T00:00:00.000000000000040", ""),
         (
             datetime(1, 1, 1, ys=40),
             "0001-01-01T00:00:00.000000000000000000000040",
+            "",
         ),
         (
             datetime(2020, 4, 20, 15, 10, 33, 976508, 569718000, 529850102),
             "2020-04-20T15:10:33.976508569718000529850102",
+            "",
+        ),
+        (
+            datetime(2020, 4, 21, 15, 29, 34, tzinfo=tzinfo(hours=1)),
+            "2020-04-21T15:29:34",
+            "+01:00",
+        ),
+        (
+            datetime(2020, 4, 21, 15, 29, 34, tzinfo=tzinfo(hours=23)),
+            "2020-04-21T15:29:34",
+            "+23:00",
+        ),
+        (
+            datetime(1, 1, 1, ys=40, tzinfo=tzinfo(hours=1)),
+            "0001-01-01T00:00:00.000000000000000000000040",
+            "+01:00",
+        ),
+        pytest.param(
+            datetime(1, 1, 1, ys=40, tzinfo=tzinfo(hours=-1)),
+            "0001-01-01T00:00:00.000000000000000000000040",
+            "-01:00",
+            marks=pytest.mark.xfail(reason="https://github.com/ni/hightime/issues/52"),
+        ),
```

Comment:
#52

Reply:
I tried this but it obviously doesn't work because the date has dashes too.
```
iso_str = super().isoformat(sep)
tz_sign = "-" if "-" in iso_str else "+"
iso_strs = iso_str.split(tz_sign)
```